app/Http/Controllers/MessageController.php

```diff
@@ -5,27 +5,33 @@
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\DB;
 use App\Models\Message;
+use Illuminate\Support\Facades\Auth;
+use Symfony\Component\HttpFoundation\Response;
 
 class MessageController extends Controller
 {
     public function index(Request $request)
     {
-        $messages = DB::table('messages')->get();
-
-        return response()->json($messages);
+        $messages = Message::selectRaw('uuid, message, user_uuid, like_count, EXISTS(SELECT * FROM likes WHERE user_uuid=? && message_uuid=uuid) as like_status, created_at', [Auth::id()])->orderBy('created_at', 'DESC')->get();
```

Comment:
[Laravel 8.42 で加わった withExists() を試す](https://zenn.dev/nshiro/articles/cf12910e8935ad)

今回はこいつの出番かも？ 
`withCount()` の存在は知ってたんですが， `withExists()` もいつの間にか増えてましたｗ
（勘でググったら出てきた） 

---

まずベースとなるリレーションをモデルに定義しておきます。
（今回あまりリレーション機能を使っていないように見えますが， Eloquent を使うならリレーションやスコープを使わなくと少し勿体ない気もしますね）

```php
class Message extends Model
{
    public function likes(): HasMany // Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(Like::class, 'message_uuid'); // 外部キーが message_id ではない場合は第2引数必須
    }
}
```

リレーションを定義した上で，以下のようにするときれいだと思います。結果もしっかりモデルのインスタンスで取得できます。(`Builder` は `Illuminate\Database\Eloquent\Builder`)

```suggestion
        $messages = Message::query()
            ->withExists('likes as like_status', fn (Builder $query) => $query->where('user_uuid', Auth::id()))
            ->orderByDesc('created_at')
            ->get();
```

（動作確認してなくてエアプなので細かいところは確認しながらお願いします）

Reply:
VScode上だと文法エラー吐かれますが動くとかいう、謎現象起こされてます。

Reply:
PHP 7.4 の Arrow Function 対応のせいですかねぇ…
最新版のプラグインではさすがに対応されてるはず